### PR TITLE
Update Release guide to current practices

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This updates the release guide to follow what I currently do. It also updates the helper scripts to clean up some minor whitespace issues that I always need to fix before committing.

Note that this is wrapped at 88, so may have to wait for PR #24893 to go in.

## PR Checklist

**Documentation and Tests**
- [n/a] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [n/a] New plotting related features are documented with examples.

**Release Notes**
- [n/a] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [n/a] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [n/a] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]
modulo waiting for the line length to be relaxed.

File: doc/devel/release_guide.rst [hunk 1]
```diff
@@ -35,25 +35,15 @@ In addition the following test should be run and manually inspected::
...
 
-
 We automatically extract GitHub issue, PRs, and authors from GitHub via the
-API.  Copy the current :file:`doc/users/github_stats.rst` to
-:file:`doc/users/prev_whats_new/github_stats_X.Y.Z.rst`, changing the link
-target at the top of the file, and removing the "Previous GitHub Stats" section
-at the end.
+API. Copy the current :file:`doc/users/github_stats.rst` to
+:file:`doc/users/prev_whats_new/github_stats_{X}.{Y}.{Z}.rst`, changing the link target
+at the top of the file, and removing the "Previous GitHub Stats" section at the end.
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply doc/devel/release_guide.rst:46]: There is an implied 'while' before the 'changing'.

[the remark this answers is not in the document]
Comment (QuLogic) [reply doc/devel/release_guide.rst]: Sure

**Comment by QuLogic** [comment]
Note for later: add instruction to check spelling in GitHub stats, re #25084

**Comment by ksunden** [comment]
Rebased and fixed the one remaining flake8 error (docstring in `tools/github_stats.py`)

Should be good to merge upon CI passing. (Was waiting for the line length bump)

**Pull request merged**